Src/FluentAssertions/AssertionExtensions.cs

```diff
@@ -916,7 +919,25 @@ public static TaskCompletionSourceAssertions<T> Should<T>(this TaskCompletionSou
     /// <exception cref="ArgumentNullException">Thrown if <paramref name="eventSource"/> is Null.</exception>
     public static IMonitor<T> Monitor<T>(this T eventSource, Func<DateTime> utcNow = null)
     {
-        return new EventMonitor<T>(eventSource, utcNow ?? (() => DateTime.UtcNow));
+        return Monitor(eventSource, o =>
+        {
+            o.ConfigureTimestampProvider(utcNow);
+        });
+    }
+
+    /// <summary>
+    /// Starts monitoring <paramref name="eventSource"/> for its events using the given <paramref name="configureOptions"/>.
+    /// </summary>
+    /// <param name="eventSource">The object for which to monitor the events.</param>
+    /// <param name="configureOptions">
+    /// Options to configure the EventMonitor.
+    /// </param>
+    /// <exception cref="ArgumentNullException">Thrown if <paramref name="eventSource"/> is Null.</exception>
+    public static IMonitor<T> Monitor<T>(this T eventSource, Action<EventMonitorOptions> configureOptions)
+    {
+        var options = new EventMonitorOptions();
+        configureOptions?.Invoke(options);
```

Comment:
In most (all?) other assertions we call `Guard.ThrowIfArgumentIsNull(configureOptions, nameof(configureOptions));` instead of allowing a null value.

Reply:
fixed